### DISCUSSION TITLE
Add `Svc/CmdSequencer` timeout update command and command to ignore command errors

### DISCUSSION THREAD

**Discussion by timcanham** [nasa/fprime#1112] [Call for Comments]
Import commands from a previous project to override the default sequencer timeout and to optionally ignore command failure responses.

1) Update timeout value - The sequencer instance is given a timeout when it is instantiated in the topology:

```c++
      void setTimeout(
          const NATIVE_UINT_TYPE seconds //!< The number of seconds
      );
```

This value is `0` by default if `setTimeout()` is not called, which means it will block indefinitely waiting for command completions to come back from components.

This command will modify the stored timeout, which will become effective on the next command.

```xml
    <command kind="async" opcode="7" mnemonic="CS_SET_TIMEOUT">
        <comment>Override default timeout value</comment>
        <args>
            <arg type="U32" name="timeout">
                <comment>New timeout value</comment>
            </arg>
        </args>
    </command>
```

This command can be useful if a default timeout is adequate for almost all commands with some commands that are known to take longer. These commands can be surrounded by `CS_SET_TIMEOUT` commands to temporarily override the value.

**Comment by timcanham** [comment]
2. Ignore command failures - The sequencer will normally terminate the running sequence when a component returns a command execution error. There are cases where a command failure status can be ignored and the sequence can safely continue executing.

```xml
    <command kind="async" opcode="8" mnemonic="CS_SET_ERROR_MODE">
        <comment>Set the sequencer command error handling mode</comment>
        <args>
            <arg type="ENUM" name="resp">
                <comment>What the response should be</comment>
                <enum name="SeqErrResp">
                    <item name="SEQ_ERR_RESP_FAIL"/>
                    <item name="SEQ_ERR_RESP_IGNORE"/>
                </enum>
            </arg>
        </args>
    </command>
```

A command that may safely fail can be surrounded by the command to `SEQ_ERR_RESP_IGNORE` and then to `SEQ_ERR_RESP_FAIL` to restore the error handling.

**Reply by LeStarch** [reply]
Some of my projects have a driving need for this feature.

**Comment by SterlingPeet** [comment]
Is there a reason that neither of these are parameters?  Is the only use case for adjusting the default, temporary?

**Reply by timcanham** [reply]
The use case we have imagined for it is to embed the commands in a sequence to work around particular commands. It could be a parameter too if there is enough community demand, but trying to follow the KISS principle for now.

**Reply by SterlingPeet** [reply]
Fair, KISS is a perfectly valid justification in my view.
